lib/ansible/modules/network/nxos/nxos_tms_destgroup.py

```diff
@@ -0,0 +1,217 @@
+        cmd_ref._ref['destination']['playval']['protocol'] = protocol
+        cmd_ref._ref['destination']['playval']['encoding'] = encoding
+    if cmd_ref._ref.get('destination').get('existing'):
+        for key in cmd_ref._ref['destination']['existing'].keys():
+            protocol = cmd_ref._ref['destination']['existing'][key]['protocol'].lower()
+            encoding = cmd_ref._ref['destination']['existing'][key]['encoding'].lower()
+            cmd_ref._ref['destination']['existing'][key]['protocol'] = protocol
+            cmd_ref._ref['destination']['existing'][key]['encoding'] = encoding
+
+    return cmd_ref
+
+
+def aggregate_input_validation(module, item):
+    if 'identifier' not in item:
+        msg = "aggregate item: {0} is missing required 'identifier' parameter".format(item)
+        module.fail_json(msg=msg)
+    if 'destination' in item and not isinstance(item['destination'], dict):
+        msg = "aggregate item: {0} parameter 'destination' must be a dict".format(item)
+        msg = msg + " defining values for keys: ip, port, protocol, encoding"
+        module.fail_json(msg=msg)
+    if 'destination' not in item and len(item) > 1:
+        msg = "aggregate item: {0} contains unrecognized parameters.".format(item)
+        msg = msg + " Make sure 'destination' parameter keys are specified as follows"
+        msg = msg + " destination: {ip: <ip>, port: <port>, protocol: <prot>, encoding: <enc>}}"
+        module.fail_json(msg=msg)
+
+
+def get_aggregate_cmds(module):
+    ''' Get list of commands from aggregate parameter '''
+
+    cache_existing = None
+    proposed_cmds = []
+    aggregate = module.params.get('aggregate')
+    for item in aggregate:
+        aggregate_input_validation(module, item)
+        for k, v in item.items():
+            if 'identifier' in k:
+                module.params['identifier'] = v
+            if 'destination' in k:
+                module.params['destination'] = {}
+                for k, v in v.items():
+                    if 'ip' in k:
+                        module.params['destination']['ip'] = v
+                    if 'port' in k:
+                        module.params['destination']['port'] = v
+                    if 'protocol' in k:
+                        module.params['destination']['protocol'] = v
+                    if 'encoding' in k:
+                        module.params['destination']['encoding'] = v
+        resource_key = 'destination-group {0}'.format(module.params['identifier'])
+        cmd_ref = NxosCmdRef(module, TMS_CMD_REF)
+        cmd_ref.set_context([resource_key])
+        if cache_existing:
+            cmd_ref.get_existing(cache_existing)
+        else:
+            cmd_ref.get_existing()
+            cache_existing = cmd_ref.cache_existing
+        cmd_ref.get_playvals()
+        cmd_ref = normalize_data(cmd_ref)
+        cmds = cmd_ref.get_proposed()
+        proposed_cmds.extend(cmds)
```

Comment:
I'm okay with the code above, just wondering if it would be good to disambiguate the multi-instance `cmd_ref` objects in this method's for loop from the single-instance `cmd_ref` in `main()` ?  
e.g. make these `mi_cmd_ref`, `agg_cmd_ref`, etc.

Your call.

Reply:
Nah.. I plan to leave it as is.  It's in a separate method so I think it's clear enough and besides the cmd_ref objetcs in  `get_aggregate_cmds` are discarded and only the resulting command set is returned.

Reply:
👍 
yup, just discombobulating the combobulated.